File: tests/fixtures/sample_kotlin/service.kt
package app.service

import app.models.User

// A top-level function (Kotlin allows file-level functions).
fun validate(name: String): Boolean {
    return name.isNotEmpty()
}

class AuthService {
    // `u: User` is an explicitly declared receiver type, so `u.isActive()` resolves.
    fun authenticate(u: User): Boolean {
        return validate("admin") && u.isActive()
    }

    // `User("admin")` is a constructor call (Kotlin has no `new`).
    fun make(): User {
        return User("admin")
    }
}

File: tests/fixtures/sample_swift/Service.swift
// A top-level function (resolved repo-wide — Swift's flat module namespace needs no import).
func validate(name: String) -> Bool {
    return name.isEmpty == false
}

class AuthService {
    // `u: User` is an explicitly declared receiver type, so `u.isActive()` resolves.
    func authenticate(u: User) -> Bool {
        return validate(name: "admin") && u.isActive()
    }

    // `User(name:)` is an initializer call (Swift has no `new`).
    func make() -> User {
        return User(name: "admin")
    }
}
